platform-api/src/main/java/io/datacater/core/config/ConfigUtilities.java

```diff
@@ -0,0 +1,118 @@
+package io.datacater.core.config;
+
+import io.datacater.core.authentication.DataCaterSessionFactory;
+import io.datacater.core.config.enums.Kind;
+import io.datacater.core.deployment.DeploymentSpec;
+import io.datacater.core.exceptions.IncorrectConfigException;
+import io.datacater.core.stream.Stream;
+import io.datacater.core.utilities.JsonUtilities;
+import io.smallrye.mutiny.Uni;
+import java.util.*;
+
+public class ConfigUtilities {
+
+  public static Uni<List<ConfigEntity>> getConfig(
+      List<String> configs, DataCaterSessionFactory dsf) {
+    return dsf.withTransaction(
+        (session, transaction) ->
+            session
+                .createQuery("from ConfigEntity", ConfigEntity.class)
+                .getResultList()
+                .onItem()
+                .transform(
+                    x -> x.stream().filter(item -> configs.contains(item.getName())).toList())
+                .onItem()
+                .ifNull()
+                .continueWith(new ArrayList<>()));
+  }
+
+  public static List<String> getConfigNames(Map<String, List<String>> labels) {
```

Comment:
I feel that the function name should indicate that it is searching the labels and not the names of the configs.

It looks like it is always returning one name, so maybe make the name singular?

Reply:
How does `extractConfigNames()` or `extractConfigNamesFromLabels()` sound?

> I feel that the function name should indicate that it is searching the labels and not the names of the configs.

The method takes a list of labels, searches them and returns a list of config names extracted from the labels.
My naming logic is "What does the method do -> getConfigNames" with the "extracted from labels" being implicit due to the method parameter.

> It looks like it is always returning one name, so maybe make the name singular?

Lists are possible

Reply:
My fault - got it! I believe that you can also close this conversation and keep it as it is :)